pkg/scheduler/framework/plugins/dynamicresources/allocateddevices.go

```diff
@@ -66,14 +79,18 @@ func foreachAllocatedDevice(claim *resourceapi.ResourceClaim, cb func(deviceID s
 type allocatedDevices struct {
 	logger klog.Logger
 
-	mutex sync.RWMutex
-	ids   sets.Set[structured.DeviceID]
+	mutex      sync.RWMutex
+	ids        sets.Set[structured.DeviceID]
+	shareIDs   sets.Set[structured.SharedDeviceID]
+	capacities structured.ConsumedCapacityCollection
 }
 
 func newAllocatedDevices(logger klog.Logger) *allocatedDevices {
 	return &allocatedDevices{
-		logger: logger,
-		ids:    sets.New[structured.DeviceID](),
+		logger:     logger,
+		ids:        sets.New[structured.DeviceID](),
+		shareIDs:   sets.New[structured.SharedDeviceID](),
+		capacities: make(map[structured.DeviceID]structured.ConsumedCapacity),
```

Comment:
```suggestion
		capacities: make(structured.ConsumedCapacityCollection),
```
Or just `structured.NewConsumedCapacityCollection()`